kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/cache/Reflector.java

```diff
@@ -48,18 +47,18 @@
   private final AtomicBoolean isWatcherStarted;
   private final AtomicReference<Watch> watch;
 
-  public Reflector(Class<T> apiTypeClass, ListerWatcher<T, L> listerWatcher, Store store, OperationContext operationContext, long resyncPeriodMillis) {
+  public Reflector(Class<T> apiTypeClass, ListerWatcher<T, L> listerWatcher, Store store, OperationContext operationContext, long resyncPeriodMillis, ScheduledExecutorService resyncExecutor) {
     this.apiTypeClass = apiTypeClass;
     this.listerWatcher = listerWatcher;
     this.store = store;
     this.operationContext = operationContext;
     this.resyncPeriodMillis = resyncPeriodMillis;
```

Comment:
Given the fact that resync period is expected to be positive, maybe we should add a check+IllegalArgumentException here so that users won't remain clueless in case for some reason this class was instantiated directly (not through Controller.java)

Reply:
ohk, right now we were using a Default resync period of 5 seconds upon invalid resync period. But throwing IllegalArgumentException sounds better to me. Let me update